firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvent.kt

```diff
@@ -29,23 +31,10 @@ internal data class SessionEvent(
 
   /** Information about the session triggering the event. */
   val sessionData: SessionInfo,
-) {
-  companion object {
-    fun sessionStart(sessionDetails: SessionDetails) =
-      SessionEvent(
-        eventType = EventType.SESSION_START,
-        sessionData =
-          SessionInfo(
-            sessionDetails.sessionId,
-            sessionDetails.firstSessionId,
-            sessionDetails.sessionIndex,
-          ),
-      )
-  }
-}
+)
 
 /** Enum denoting all possible session event types. */
-internal enum class EventType(val number: Int) {
+internal enum class EventType(override val number: Int) : NumberedEnum {
```

Comment:
Why `override`?

Reply:
This is some pretty nice Kotlin syntax. Imagine it like the interface were Java and had a method `int getNumber();` then this enum implemented it and overrode the getter method and returned the member variable.